Add PBS qsub self submission lib

Kirill888 opened the pull request:
This is cleaned up copy from datacube stats with tests added.

Kirill888 committed:
Add PBS qsub self submission lib

This is cleaned up copy from datacube stats with tests added.

jeremyh commented on digitalearthau/qsub.py:
```diff
@@ -0,0 +1,636 @@
+def generate_self_launch_script(*args):
+    s = "#!/bin/bash\n\n"
+    s += pbs.generate_env_header()
+    s += '\n\nexec ' + ' '.join("'{}'".format(s) for s in self_launch_args(*args))
```
There's a handy built-in for quoting shell args (that'll handle other special characters too): `shlex.quote(arg)`

jeremyh commented on scripts/qsub_sample_app.py:
```diff
@@ -0,0 +1,108 @@
+        yield Task(i)._asdict()
+
+
+# pylint: disable=invalid-name
```
Why did this need to be disabled? Might be worth adding a comment above it

jeremyh reviewed (approved):
Thanks Kirill

Kirill888 committed:
Address code review - safer shell escapes

- Using shlex.quote for safer handling of shell escapes
- explain pylint disable in the test app
